packages/hdwallet-native/src/ethereum.ts

```diff
@@ -38,46 +39,52 @@ export function MixinNativeETHWalletInfo<TBase extends core.Constructor>(Base: T
   };
 }
 
-export function MixinNativeETHWallet<TBase extends core.Constructor>(Base: TBase) {
+export function MixinNativeETHWallet<TBase extends core.Constructor<NativeHDWalletBase>>(Base: TBase) {
   return class MixinNativeETHWallet extends Base {
     _supportsETH = true;
 
     #ethWallet: Wallet;
 
     ethInitializeWallet(seed: string): void {
-      this.#ethWallet = new Wallet(HDNode.fromSeed(seed).derivePath(defaultPath));
+      return this.needsMnemonic(!!seed, () => {
+        this.#ethWallet = new Wallet(HDNode.fromSeed(seed).derivePath(defaultPath));
+      });
     }
 
     async ethGetAddress(msg: core.ETHGetAddress): Promise<string> {
-      return this.#ethWallet.getAddress();
+      return this.needsMnemonicAsync(!!this.#ethWallet, () => this.#ethWallet.getAddress());
```

Comment:
```suggestion
      return this.needsMnemonicAsync(!!this.#ethWallet, async () => this.#ethWallet.getAddress());
```

Reply:
The anonymous function doesn't need to be marked `async` because `getAddress` is async. That's why TypeScript isn't giving an error here because the return type is Promise.

Reply:
Yeah that makes sense. It "looks" more right in ref to `needsMnemonicAsync` without having to know `getAddress` return type. I guess like you said typescript would complain if need be. Nitpick for looks overall. Your call.